### DISCUSSION TITLE
xarray.Dataset to mikeio.Dataset

### DISCUSSION THREAD

**Discussion by lezhongwen** [DHI/mikeio#386] [Q&A]
I'm helping my friend with geographical data.  My job is to convert unstructured data into structured data in DFS0, DFS1, DFS2, DFS3 and DFSU formats.
In using it, I found that mikeio can convert mikeio.Dataset to xarray.Dataset. I am sorry for that I am not familiar with the details of DFS files. Comes the question,  is it possible to add an API for converting xarray. Dataset to the mikeio.Dataset type?  If it works, this API will help us a lot!
Thanks a lot again for your contribution!

**Comment by ecomodeller** [comment]
xarrray is a library for multidimensional arrays (structured data), what kind of unstructured data do are you trying to convert to dfs?

**Reply by lezhongwen** [reply]
I got the data written in txt file from my partner. I have built an xarray.Dataset for the data, but I don't know how to change them into mikeio.Dataset.

**Reply by ecomodeller** [reply]
It is not possible to convert a xarray.Dataset to mikeio.Dataset in general, since xarray is more flexible than is supported by dfs files. If your end goal is a dfs file, and you have data in txt, it is a better idea to take a look at the examples where Dfs files are created from text files e.g. https://dhi.github.io/getting-started-with-mikeio/dfs0.html#writing-data

**Reply by lezhongwen** [reply]
Thank you for your reply. Initially, I thought building structured data using xarray would be friendlier to developers who were new to your library, so I came up with the idea. By reading your code, I found that it is also very convenient to build a mikeio.Dataset directly. Thanks again for your contribution!
